webapp/src/sui.tsx

```diff
@@ -90,55 +91,87 @@ export class UiElement<T extends WithPopupProps> extends data.Component<T, {}> {
 }
 
 export class DropdownMenuItem extends UiElement<DropdownProps> {
+    private isOpened = false
+    private preventHide = false
+
+    private handleTab = (e: KeyboardEvent) => {
+        let charCode = (typeof e.which == "number") ? e.which : e.keyCode
+        if (charCode === 9) {
+            this.close()
+        } else if (charCode === 32 || charCode === 13) {
+            /* give the focus back to the dropdown menu, so if the menuitem opens a modal,
+               the focus will not be reset once the modal is closed. */
+            this.child("").focus()
+        }
+    }
+
+    private close() {
+        this.preventHide = false
+        this.child("").dropdown("hide")
+    }
+
     componentDidMount() {
         this.popup()
         this.child("").dropdown({
-            action: (text: string, value: any, element: any) => {
-                let htmlElement: HTMLElement
+            action: (text: string, value: any, element: JQuery) => {
```

Comment:
As now, when we navigate with the keyboard, the items have the focus, we don't need to simulate a click on them as by default they have the "fireClickOnEnter". But there was still a small problem with the "a href" tags. With this new implementation, it works with Enter key and Click, and the Privacy menu is not opened 2 times on Edge and Safari anymore.